lib/src/bridge/characteristics_mixin.dart

```diff
@@ -66,11 +77,16 @@ mixin CharacteristicsMixin on FlutterBLE {
           )
           .catchError((errorJson) =>
               Future.error(BleError.fromJson(jsonDecode(errorJson.details))))
-          .then(
-            (rawJsonValue) =>
-                _parseCharacteristicWithValueWithTransactionIdResponse(
-                    peripheral, rawJsonValue),
-          );
+          .then((rawValue) {
+            String rawJsonValue = "";
+            if (rawValue is String) {
+              rawJsonValue = rawValue;
+            }
```

Comment:
This can actually be get rid of by changing to `.invokeMethod<String>(...)`
And yes, it's guaranteed to be string at this point in time.